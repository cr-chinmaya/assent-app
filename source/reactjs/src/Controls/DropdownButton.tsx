import * as React from 'react';
import { DefaultButton } from '@fluentui/react';
import { IContextualMenuProps } from '@fluentui/react/lib/ContextualMenu';

const DropdownButton = ({
    text = '',
    title = '',
    menuItems = [],
    primary = false,
    disabled = false,
    className = ''
}: {
    text: string;
    title: string;
    menuItems?: any;
    primary?: boolean;
    disabled?: boolean;
    className: string;
}): any => {
    const menuProps: IContextualMenuProps = {
        items: menuItems
    };
    return (
        <DefaultButton
            className={className}
            text={text}
            title={title}
            menuProps={menuProps}
            primary={primary}
            disabled={disabled}
        />
    );
};

export default DropdownButton;
