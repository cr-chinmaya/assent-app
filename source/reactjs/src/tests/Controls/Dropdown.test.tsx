import * as React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import 'jest-styled-components';
import DropdownButton from '../../Controls/DropdownButton';

describe('DropdownButton', () => {
    it('DropdownButton renders correctly', () => {
        const wrapper = render(<DropdownButton />);
        expect(wrapper).toMatchSnapshot();
    });
    it('DropdownButton renders correctly', () => {
        const menuitem = ['Op1', 'Op2'];
        const wrapper = render(
            <DropdownButton
                className=''
                menuItems={menuitem}
                text={'Test'}
                title={'Test Drop down Button'}
                primary={true}
                disabled={true}
            />
        );
        expect(wrapper.getByRole('button')).toBeTruthy;
    });
});
