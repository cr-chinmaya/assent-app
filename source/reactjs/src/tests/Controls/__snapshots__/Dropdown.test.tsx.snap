// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownButton DropdownButton renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="ms-Button ms-Button--default ms-Button--hasMenu root-40"
        data-is-focusable="true"
        title=""
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-41"
          data-automationid="splitbuttonprimary"
        >
          <span
            class="ms-Button-label label-44"
            id="id__0"
          />
          <i
            aria-hidden="true"
            class="ms-Icon root-37 ms-Button-menuIcon menuIcon-45"
            data-icon-name="ChevronDown"
            role="presentation"
          />
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="ms-Button ms-Button--default ms-Button--hasMenu root-40"
      data-is-focusable="true"
      title=""
      type="button"
    >
      <span
        class="ms-Button-flexContainer flexContainer-41"
        data-automationid="splitbuttonprimary"
      >
        <span
          class="ms-Button-label label-44"
          id="id__0"
        />
        <i
          aria-hidden="true"
          class="ms-Icon root-37 ms-Button-menuIcon menuIcon-45"
          data-icon-name="ChevronDown"
          role="presentation"
        />
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
